src/main/java/oncall/domain/WorkerOrder.java

```diff
@@ -0,0 +1,52 @@
+package oncall.domain;
+
+import oncall.domain.iterator.LoopIterator;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public final class WorkerOrder {
+    private static final int MIN_WORKER_COUNT = 5;
+    private static final int MAX_WORKER_COUNT = 35;
+
+    private final List<Worker> workers;
+
+    public WorkerOrder(List<Worker> workers) {
+        this.workers = new ArrayList<>(workers);
+        validate();
+    }
+
+    private void validate() {
+        validateWorkerCount();
+        validateDuplicateWorker();
+    }
+
+    private void validateWorkerCount() {
+        if (workers.size() < MIN_WORKER_COUNT || workers.size() > MAX_WORKER_COUNT) {
+            throw new IllegalArgumentException(
+                    String.format("[ERROR] 근무자는 %d명 이상 %d명 이하로 입력해주세요.", MIN_WORKER_COUNT, MAX_WORKER_COUNT));
+        }
+    }
+
+    private void validateDuplicateWorker() {
+        if (workers.stream().distinct().count() != workers.size()) {
```

Comment:
```suggestion
        int distinctSize = (int) workers.stream()
                .distinct()
                .count();
        if(distinctSize != workers.size()) {
```

저는 개인적으로 조건문 내부를 최대한 단순하게 구성하려 노력하는 편입니다.
때문에 특정 값을 생성해 확인해야 하는 경우 외부에서 값을 생성해 주고 조건문에서는 이미 생성된 값을 사용하기만 하는 것을 선호하며, 이 경우 가독성이 더 좋아진다고 생각합니다.
물론 개인 취향의 영역일 것 같기는 합니다!

Reply:
개인 취향이라면 개인 취향이라 할 수 있지만, 가독성을 높이는 것을 늘 고민하는 것은 좋은 것 같아요 :)

저 또한 짱수처럼 하는게 훨씬 가독성이 좋아 보여요!!